### PR TITLE
ROX-17965: Create separate logs for specs

### PR DESCRIPTION
**Pull request by gavin-stackrox**
## Description

This PR create a separate log file for each groovy test specification so that it is possible to view their operation in a coherent manner. Without this change the only log available is the combined build.log which interleaves the output from each test thread as they happen. This change is demoed via https://github.com/stackrox/stackrox/pull/7026.

## Why not have this handled by Gradle/Junit/Spock/Wotsit?

While Junit does support separation of output by thread ([ref](https://junit.org/junit5/docs/current/user-guide/#running-tests-capturing-output)) via TestExecutionListener, this does not appear to be supported by Gradle: https://github.com/gradle/gradle/issues/4605, https://github.com/junit-team/junit5/issues/2115. It might be possible if the tests used Junit's jupiter engine but I think that is separate from the spock junit engine (I'm on shaky ground here).

## Checklist
- [x] Investigated and inspected CI test results

## Testing Performed

CI is sufficient.

### CODE REVIEW

File: qa-tests-backend/src/test/groovy/BaseSpecification.groovy [hunk 1]
```diff
@@ -226,6 +227,8 @@ class BaseSpecification extends Specification {
     private long testSpecStartTimeMillis
 
     def setupSpec() {
+        MDC.put("logFileName", this.class.getSimpleName())
+        MDC.put("specification", this.class.getSimpleName())
         log.info("Starting testsuite")
```
Comment (janisz) [inline qa-tests-backend/src/test/groovy/BaseSpecification.groovy:232]: Why do we need it in 2 places?

Comment (gavin-stackrox) [reply qa-tests-backend/src/test/groovy/BaseSpecification.groovy:232]: > Why do we need it in 2 places?

I couldn't figure it out but without the extra `.put()` in the `setup()` the keys were cleared. I'll add a comment for that.

**Review by janisz — APPROVED** [review]

**Review by janisz — APPROVED** [review]

**Pull request merged**